src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,365 @@
+        """
+        query_embedding = self.embedding.embed_query(query)
+        return self.max_marginal_relevance_search_by_vector(
+            query_embedding,
+            k=k,
+            fetch_k=fetch_k,
+            lambda_mult=lambda_mult,
+            filters=filters,
+        )
+
+    def max_marginal_relevance_search_by_vector(
+        self,
+        embedding: List[float],
+        k: int = 4,
+        fetch_k: int = 20,
+        lambda_mult: float = 0.5,
+        filters: Optional[BaseFilter] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        """Run max marginal relevance search on the results of Firestore nearest
+        neighbor search using a vector. This method will throw if the index is
+        not created, in which case you will be prompted to create the index.
+
+        Raises:
+            FailedPrecondition: If the index is not created.
+
+        Args:
+            embedding: The query vector.
+            k: The number of documents to return. Defaults to 4.
+            fetch_k: The number of documents to fetch. Defaults to 20.
+            lambda_mult: The lambda multiplier. Defaults to 0.5.
+            filters: The pre-filter to apply to the query. Defaults to None.
+
+        Returns:
+            List[Document]: List of documents most similar to the query vector.
+        """
+        doc_results = self._similarity_search(embedding, fetch_k, filters=filters)
+        doc_embeddings = [
+            self._vector_to_list(d.to_dict()[self.embedding_field]) for d in doc_results
+        ]
+        mmr_doc_indexes = maximal_marginal_relevance(
+            np.array(embedding, dtype=np.float32),
+            doc_embeddings,
+            lambda_mult=lambda_mult,
+            k=k,
+        )
+        return [convert_firestore_document(doc_results[i]) for i in mmr_doc_indexes]
+
+    @classmethod
+    def from_texts(
+        cls: Type[FirestoreVectorStore],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        collection: Optional[str | CollectionReference] = None,
+        **kwargs: Any,
+    ) -> FirestoreVectorStore:
+        """Create a FirestoreVectorStore instance and add texts to it.
+
+        Args:
+            texts: The texts to add to the vector store.
+            embedding: The embeddings to use to generate the vectors from texts.
+            metadatas: The metadata to add to the vector store. Defaults to None.
+
+        Returns:
+            FirestoreVectorStore: The FirestoreVectorStore instance.
+        """
+        if collection is None:
+            raise ValueError("Must provide 'collection' named parameter.")
+
+        vs_obj = cls(collection=collection, embedding=embedding, **kwargs)
+        vs_obj.add_texts(texts, metadatas, ids, **kwargs)
+        return vs_obj
+
+    def _select_relevance_score_fn(self) -> Callable[[float], float]:
```

Comment:
Where is this method called? -- I don't see references for this method.

Reply:
This was a left-over from the base class, it's used by `similarity_search_with_score` but in our implementation since the nearest neighnours search doesn't return scores we won't implmenet it, removing.